financial-templates-lib/clients/ExpiringMultiPartyClient.js

```diff
@@ -153,25 +149,24 @@ class ExpiringMultiPartyClient {
     this.expiredLiquidations = expiredLiquidations;
     this.disputedLiquidations = disputedLiquidations;
 
-    this.positions = this.sponsorAddresses.reduce(
-      (acc, address, i) =>
-        // Filter out empty positions.
-        positions[i].rawCollateral.toString() === "0"
-          ? acc
-          : /* eslint-disable indent */
-            acc.concat([
-              {
-                sponsor: address,
-                withdrawalRequestPassTimestamp: positions[i].withdrawalRequestPassTimestamp,
-                withdrawalRequestAmount: positions[i].withdrawalRequestAmount.toString(),
-                numTokens: positions[i].tokensOutstanding.toString(),
-                amountCollateral: collateral[i].toString(),
-                hasPendingWithdrawal: positions[i].withdrawalRequestPassTimestamp > 0
-              }
-            ]),
-      []
-    );
-    this.lastUpdateTimestamp = await this.emp.methods.getCurrentTime().call(undefined, this.latestBlock);
+    this.positions = this.sponsorAddresses.reduce((acc, address, i) => {
+      const rawCollateral = this.toBN(positions[i].rawCollateral.toString());
+      // Filter out empty positions.
+      return rawCollateral.isZero()
+        ? acc
+        : /* eslint-disable indent */
+          acc.concat([
+            {
+              sponsor: address,
+              withdrawalRequestPassTimestamp: positions[i].withdrawalRequestPassTimestamp,
+              withdrawalRequestAmount: positions[i].withdrawalRequestAmount.toString(),
+              numTokens: positions[i].tokensOutstanding.toString(),
+              amountCollateral: rawCollateral.mul(this.cumulativeFeeMultiplier).div(this.toBN(this.toWei("1"))),
```

Comment:
this is great! I had seen the same optimization: there is no point in calling `this.emp.methods.getCollateral` as it simply calculates this same value. Good catch @nicholaspai

Reply:
one thing is that I _think_ you need to convert this to a string for it to be correct. The rest of this like should stay the same but just cast it at the end with a `.toString()`.

Reply:
Yeah this failed the unit tests and I just changed